Userland/Utilities/network-settings.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2023, Fabian Dellwing <fabian@dellwing.net>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/JsonObject.h>
+#include <AK/JsonParser.h>
+#include <LibCore/ConfigFile.h>
+#include <LibCore/System.h>
+#include <LibMain/Main.h>
+#include <unistd.h>
+
+ErrorOr<int> serenity_main(Main::Arguments)
+{
+    TRY(Core::System::pledge("stdio rpath wpath cpath recvfd sendfd proc exec"));
+
+    TRY(Core::System::unveil("/bin/NetworkServer", "x"));
+    TRY(Core::System::unveil("/etc/Network.ini", "rwc"));
+    TRY(Core::System::unveil(nullptr, nullptr));
+
+    auto infile = TRY(Core::File::standard_input());
+
+    auto input_bytes = TRY(infile->read_until_eof());
+    StringView json_data = input_bytes;
+
+    if (json_data.is_empty() || json_data.is_whitespace())
+        return Error::from_errno(EINVAL);
+
+    auto json = TRY(JsonParser(json_data).parse());
+
+    if (!json.is_object())
+        return Error::from_errno(EINVAL);
+
+    auto json_object = json.as_object();
+
+    auto config_file = TRY(Core::ConfigFile::open_for_system("Network", Core::ConfigFile::AllowWriting::Yes));
+    json_object.for_each_member([&](DeprecatedString const& adapter_name, JsonValue const& adapter_data) {
+        adapter_data.as_object().for_each_member([&](const DeprecatedString& key, const JsonValue& value) {
+            switch (value.type()) {
+            case JsonValue::Type::String:
+                config_file->write_entry(adapter_name, key, value.as_string());
+                break;
+            case JsonValue::Type::Bool:
+                config_file->write_bool_entry(adapter_name, key, value.as_bool());
+                break;
+            case JsonValue::Type::Null:
+                break;
+            default:
+                dbgln("Extend switch/case key={}, value={}", key, value);
+                VERIFY_NOT_REACHED();
+            }
+        });
+    });
+    TRY(config_file->sync());
+
+    auto command = Vector<StringView>();
+    TRY(command.try_append("/bin/NetworkServer"sv));
+    TRY(Core::System::exec_command(command, true));
```

Comment:
I will leave a FIXME, but it will be quite different. Because even though it is called `NetworkServer` it is in no way something that should be called "server" right now. It is a binary that is executed and exits after it did its thing.

Reply:
Oh, you're right, it does indeed exit! That's weird, I thought it would take care of DHCP leases, but in order to do that it has to send network packets every 24 hours … I guess nobody ever runs Serenity for long enough to notice this.

Reply:
`DHCPClient` runs as deamon in the background if I remember correctly.